### ISSUE TITLE
pass NEXTCLOUD_VERSIONS the command line instead

### ISSUE THREAD

**Issue by teto** [helsinki-systems/nc4nix#8]
I've recently tried to improve my nextcloud setup (including a full reinstall, here are some notes https://discourse.nixos.org/t/how-to-get-beyond-the-current-nextcloud-module/30533/3).

I was looking at patching https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/nextcloud/packages/generate.sh  (to be able to run it from any directory) and was confused by the NEXTCLOUD_VERSIONS variable that did not seem to be used ? Is there a way to pass the nextcloud version on the CLI instead so that it's more explicit ?

Also I wonder if you had considered moving the repo to https://github.com/nix-community/ ? many *2nix programs live there to ease contributing/discovery.

Also would be nice to have a `--help` available, if only to discover how to use the program.

**Comment by teto** [comment]
yeah I am not surprised about nix-community but other points remain valid:
I've opened a PR to make the updater more usable but I am kinda limited by nc4nix, would be nice to give it an output directory as well see:
https://github.com/NixOS/nixpkgs/pull/244057#issuecomment-1638949136

Also it's slow as hell, could the update script run in parallel several nc4nix, one per nextcloud version ?

**Comment by onny** [comment]
> yeah I am not surprised about nix-community but other points remain valid: I've opened a PR to make the updater more usable but I am kinda limited by nc4nix, would be nice to give it an output directory as well see: [NixOS/nixpkgs#244057 (comment)](https://github.com/NixOS/nixpkgs/pull/244057#issuecomment-1638949136)

Regarding the initial point of your feature request here, I'm fine with having NEXTCLOUD_VERSIONS as an environment variable. Having a PR which implements a command line argument for this would be fine too.

> Also it's slow as hell, could the update script run in parallel several nc4nix, one per nextcloud version ?

Could be a point for an other issue or feature request here :)

**Comment by teto** [comment]
To be clear I dont plan to submit PRs here (no time, dont know go) but it's annoying to not be able how the program works without looking at strace. Adding a `--help` would go a long way, even better a --output or just dump the generated code on stdout so that the user can redirect this to a file of its choice.  https://github.com/NixOS/nixpkgs/pull/244057#issuecomment-1640128428

**Comment by leonklingele** [comment]
Would adding a `How to run` section to the `README.md` with something à la `Run this app using NEXTCLOUD_VERSIONS=27,28 nc4nix` suffice here?
